commands/uploader.go

```diff
@@ -66,11 +75,42 @@ func (c *uploadContext) prepareUpload(unfiltered ...*lfs.WrappedPointer) (*tq.Tr
 		}
 		uniqOids.Add(p.Oid)
 
-		// estimate in meter early (even if it's not going into uploadables), since
-		// we will call Skip() based on the results of the download check queue.
-		c.meter.Add(p.Size)
+		// canUpload determines whether the current pointer "p" can be
+		// uploaded through the TransferQueue below. It is set to false
+		// only when the file is locked by someone other than the
+		// current committer.
+		var canUpload bool = true
+
+		lockQuery := map[string]string{"path": p.Name}
+		locks, err := c.lockClient.SearchLocks(lockQuery, 1, false)
```

Comment:
Making an API call per file seems like a lot of overhead. I think our early attempts queried all the locks up front, and checked each file locally. That does mean that we'll have to maintain a potentially huge list of locks in memory locally, so we may want to look at an improved interface for verifying groups of files in a single POST request.